actions/run/dist/index.js

```diff
@@ -2901,26 +2902,40 @@ async function run() {
     if (flags) {
         args.push(...flags.split(' '));
     }
-    if (artifact) {
-        args.push('--artifact', `${earthfile}+${target}/`, `${artifactPath}`);
-    }
-    else {
-        args.push(`${earthfile}+${target}`);
-    }
     if (targetFlags) {
         args.push(...targetFlags.split(' '));
     }
-    core.info(`Running command: ${command} ${args.join(' ')}`);
-    const output = await spawnCommand(command, args);
-    const imageOutput = parseImage(output);
-    if (imageOutput) {
-        core.info(`Found image: ${imageOutput}`);
-        core.setOutput('image', imageOutput);
-    }
-    const artifactOutput = external_path_.join(earthfile, parseArtifact(output));
-    if (artifactOutput !== earthfile) {
-        core.info(`Found artifact: ${artifactOutput}`);
-        core.setOutput('artifact', artifactOutput);
+    core.info(`Filtered targets >> ${targets}`);
+    const targetsArr = targets.split(' ');
+    for (const tg of targetsArr) {
+        // Get the filtered targets associated with the pattern target and earthfile.
+        core.info(`Pushing target ${tg}`);
+        targetsArgs.push(`${earthfile}+${tg}`);
+    }
+    // Running each target command in different process.
+    for (const t of targetsArgs) {
+        core.info(`Target: ${t}`);
+        const argsSpawn = [...args];
+        // Artifact is set
+        if (artifact) {
+            core.info(`Pushing target ${t} with artifact tag`);
+            argsSpawn.push('--artifact', `${t}/`, `${artifactPath}`);
+        }
+        else {
+            argsSpawn.push(t);
+        }
+        core.info(`Running command: ${command} ${argsSpawn.join(' ')}`);
+        const output = await spawnCommand(command, argsSpawn);
+        const imageOutput = parseImage(output);
+        if (imageOutput) {
+            core.info(`Found image: ${imageOutput}`);
+            core.setOutput('image', imageOutput);
+        }
+        const artifactOutput = external_path_.join(earthfile, parseArtifact(output));
```

Comment:
So this will be a bit of a bug. In the original design, there was only ever one target, which means only a single artifact is ever produced. Now, if two targets are run, the `artifact` output will be set to the last one that ran (which is probably confusing). 

Fixing this will be a bit of work. If we want, for now, we just need to specify in the documentation that it doesn't support handling multiple targets that each produce an artifact.

The same issue exists with images.

Reply:
Maybe just set it to all targets? Not sure how should it be.

Reply:
Current Resolution: for now wildcard should only be implemented to targets that don't produce any artifact or image. This will be mentioned in the document.